lib/Fencer/Logger.hs

```diff
@@ -0,0 +1,21 @@
+-- | A module for adjusting log level values to the title case. This
+-- is needed to make sure tinylog works with value cases used by
+-- lfyt/ratelimit.
+module Fencer.Logger (adjustCase) where
```

Comment:
This should be more principled.

lyft/ratelimit uses the logrus library for logging. It defines the following log levels:

https://github.com/sirupsen/logrus/blob/39a5ad12948d094ddd5d5a6a4a4281f453d77562/logrus.go#L25

We should use the following mapping to match logrus:

```haskell
"panic" -> Fatal -- tinylog does not have Panic
"fatal" -> Fatal
"error" -> Error
"warn" -> Warn
"warning" -> Warn
"info" -> Info
"debug" -> Debug
"trace" -> Trace
```

Next, overwriting an environment variable is not the best approach.

* I would suggest using `tinylog-0.15.0`, which provides an option called `readEnvironment`: https://hackage.haskell.org/package/tinylog-0.15.0/docs/System-Logger.html#v:readEnvironment. If you set it to False, tinylog won't read env variables by itself, so you can feed whatever you want to it.
* We can then get rid of `LOG_BUFFER` and friends (they are kinda obscure and increase the surface level of Fencer for no good reason).
* You can read the variable `LOG_LEVEL` manually, lowercase it (no need for `titlecase` anymore), and transform it to a `Level` with the snippet above.

Reply:
And you can do the LOG_LEVEL parsing in `Fencer.Settings`, so `Fencer.Logger` becomes redundant.

Reply:
Can you be more specific about what variables you had in mind in the following?

> We can then get rid of LOG_BUFFER and friends (they are kinda obscure and increase the surface level of Fencer for no good reason).

Reply:
`LOG_BUFFER`, `LOG_NETSTR`, `LOG_LEVEL_MAP`.

Reply:
In the upcoming commit I removed these 3 `LOG_` variables that you suggested.

Reply:
I implemented all of the remarks from above.